content/blog/2018/02/2018-02-22-cheetah.adoc

```diff
@@ -0,0 +1,134 @@
+
+== Yes, but what does it DO? 
+
+Project Cheetah offers several things, but the most important is Durability Settings for all Pipelines, and especially the Performance-Optimized setting.  This setting avoids several unexpected performance "surprises" that may strike users.  In the general case, it greatly reduces the disk IO needs for Pipeline.  How much?  Below is a graph of storage utilization with legacy Pipeline versions (think early 2017) and with the latest version using the Performance-Optimized mode.  These are tested on an AWS instance backed by an EBS volume provisioned with 300 IOPs. 
+
+Before and After:
+
+image:/images/post-images/2018-02-22-cheetah/io-utilization.png[role="center"]
+
+As you can see, storage utilization goes down by a _lot_.  While the exact number will vary, across the benchmark testcases *this translates to 2x to 6x the Pipeline throughput before becoming IO-bound*. This also increases stability of Jenkins masters because they will tolerate unexpected load.
+
+This comes with a major drop in CPU IOWait as well:
+
+image:/images/post-images/2018-02-22-cheetah/cpu-iowait.png[role="center"]
+
+And of course the rate at which data is written to disk and number of writes/s is also reduced:
+
+image:/images/post-images/2018-02-22-cheetah/diskio.png[role="center"]
+
+For enterprise users, timing stats often show 10-20% of normal builds is serializing the Program and writing the record of steps run ("FlowNodes") - the performance optimized durability setting will cut this to _almost nothing_ (for standard pipelines, 1/100 or less) - so builds will complete faster, especially complex ones.
+
+Please see the link:/doc/book/pipeline/scaling-pipeline/[Pipeline Scalability documentation] for deeper information on the new Durability Settings and how to use them.
+
+Also, users may see a reduction in hung Pipelines because new test utilities made it possible to identify and correct a variety of bugs.
+
+== How Do I Set Speed/Durability Settings?
+There are 3 ways to configure the durability setting:
+
+=== 1. *Globally*, you can choose a global default durability setting:
+Under "Manage Jenkins" > "Configure System", labelled "Pipeline Speed/Durability Settings".  You can override these with the more specific settings below.
+
+image:/images/post-images/2018-02-22-cheetah/global-durability.png[role="center"]
+
+=== 2. *Each Pipeline* can get a custom Durability Setting:
+This is one of the job properties located at the top of the job configuration, labelled "Custom Pipeline Speed/Durability Level."  **This overrides the global setting.**  Or, use a "properties" step - the setting will apply to the NEXT run after the step is executed (same result).
+
+image:/images/post-images/2018-02-22-cheetah/pipeline-durability-highlighted.png[role="center"]
+
+[pipeline]
+----
+// Script //
+properties([durabilityHint('PERFORMANCE_OPTIMIZED')])
```

Comment:
why won't that work in the `options` directive?

Reply:
Changes in the use of "properties" with declarative: https://jenkins.io/blog/2017/01/12/declarative-pipeline-beta-2/  

"Properties" may still work (haven't tried it) but I'm trying to encourage best-practices syntax here.

Reply:
```
pipeline {
  ...
  options {
    durabilityHint('PERFORMANCE_OPTIMIZED')
  }
  ...
}
```
...will work just fine with any job property with a `@Symbol`.

Reply:
That's what I thought, so I got surprised why it was outside in the scripted section.
"Best practice" here would be to put it inside `options`